test/unit/inabox/test-inabox-viewport.js

```diff
@@ -379,16 +379,19 @@ describes.fakeWin('inabox-viewport', {amp: {}}, env => {
     });
   });
 
-  it('should disconnect friendly listener', () => {
+  it('should disconnect friendly listener and reconnect again properly', () => {
     const unobserveFunction = sandbox.spy();
-    sandbox.stub(PositionObserver.prototype, 'observe')
-        .returns(unobserveFunction);
+    const observeFunction = sandbox.stub(PositionObserver.prototype, 'observe');
+    observeFunction.returns(unobserveFunction);
     sandbox./*OK*/stub(iframeHelper, 'canInspectWindow').returns(true);
     return binding.listenForPositionSameDomain().then(() => {
+      expect(observeFunction).to.be.calledOnce;
       expect(unobserveFunction).to.not.be.called;
       binding.disconnect();
       expect(unobserveFunction).to.be.called;
-      expect(binding.unobserveFunction_).to.be.null;
+      return binding.listenForPositionSameDomain();
```

Comment:
should you just call binding.connect?

Reply:
The function is async, but since it doesn't return anything I can't check that it finishes connecting before starting assertions. I could change the function so it passes the return value from `listenForPositionSameDomain()` (and a resolved promise for the normal case, since it depends on the host script loading which we can't check anyway, and is irrelevant when testing). What do you think?

Reply:
yep, I think that's better. If possible, we could turn `listenForPositionSameDomain` back to private.

Reply:
Done.